k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java

```diff
@@ -2719,6 +2733,113 @@ public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
 
             fragmentListener.updateMenu();
         }
+
+        categorizedMessages = getMessageCategories();
+    }
+
+    int getDateCategory(long date) {
```

Comment:
This belong as a utility method in a separate class. It can then be tested properly.